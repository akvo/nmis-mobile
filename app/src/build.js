import { SERVER_URL } from '@env';
import build_json from './build.json';

const defaultBuildParams = {
  authenticationType: ['code_assignment', 'username', 'password'],
  ...build_json,
  serverURL: SERVER_URL,
};

export default defaultBuildParams;
