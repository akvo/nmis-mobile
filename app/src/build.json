{
  "authenticationType": ["code_assignment", "username", "password"],
  "serverURL": "http://192.168.21.252:8080",
  "debugMode": false,
  "dataSyncInterval": 300,
  "errorHandling": true,
  "loggingLevel": "verbose",
  "appVersion": "1.2.0",
  "lang": "en"
}
