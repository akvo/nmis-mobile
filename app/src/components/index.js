export { default as Stack } from './Stack';
export { default as Carousel } from './Carousel';
export { default as DividerLayout } from './DividerLayout';
