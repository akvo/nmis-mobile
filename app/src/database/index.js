export * from './conn';
export * from './query';
