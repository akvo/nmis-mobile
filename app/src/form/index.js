export { default as Webform } from './Webform';
