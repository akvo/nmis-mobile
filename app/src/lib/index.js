export { default as api } from './api';
export { default as loc } from './loc';
export { default as notification } from './notification';
export { default as backgroundTask } from './background-task';
export { default as cascades } from './cascades';
