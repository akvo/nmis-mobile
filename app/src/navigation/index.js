import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { HomePage, FormActionPage, FormDataPage, GetStartedPage, AuthFormPage } from '../pages';

const Stack = createNativeStackNavigator();

const RootNavigator = () => {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="GetStarted" component={GetStartedPage} />
      <Stack.Screen name="AuthForm" component={AuthFormPage} />
      <Stack.Screen name="Home" component={HomePage} />
      <Stack.Screen name="FormAction" component={FormActionPage} />
      <Stack.Screen name="FormData" component={FormDataPage} />
    </Stack.Navigator>
  );
};

const Navigation = (props) => {
  return (
    <NavigationContainer {...props}>
      <RootNavigator />
    </NavigationContainer>
  );
};

export default Navigation;
