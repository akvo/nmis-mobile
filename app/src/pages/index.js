export { default as HomePage } from './Home';
export { default as FormActionPage } from './FormAction';
export { default as FormDataPage } from './FormData';
export { default as GetStartedPage } from './GetStarted';
export { default as AuthFormPage } from './AuthForm';
export { default as SettingsPage } from './Settings';
export { default as SettingsFormPage } from './Settings/Form';
