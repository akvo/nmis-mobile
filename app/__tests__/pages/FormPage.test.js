import React from 'react';
import renderer from 'react-test-renderer';
jest.useFakeTimers();
import { FormPage } from '../../src/pages';

describe('FormPage component', () => {
  test('should render component correctly', () => {
    const tree = renderer.create(<FormPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
