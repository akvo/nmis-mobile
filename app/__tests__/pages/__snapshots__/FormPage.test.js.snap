// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage component should render component correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9fafb",
      "flex": 1,
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
  testID="stack-container"
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "backgroundColor": "#f9fafb",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
        testID="stack-container"
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View>
            <View>
              <View
                style={
                  {
                    "borderBottomWidth": 0.5,
                    "borderColor": "grey",
                    "borderTopWidth": 0.5,
                    "marginBottom": 20,
                    "paddingHorizontal": 20,
                    "paddingVertical": 15,
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#242424",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1. Introduction
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <View>
                <View>
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#242424",
                        "fontSize": 14,
                        "fontWeight": 600,
                        "marginBottom": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                    testID="field-label"
                  >
                    1. Name of the data collector (Enumerator)
                  </Text>
                  <View
                    style={
                      {
                        "paddingHorizontal": 10,
                        "width": "100%",
                      }
                    }
                    testID="RNE__Input__view-wrapper"
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomWidth": 0.5,
                          "borderColor": "grey",
                          "borderRadius": 5,
                          "borderWidth": 0.5,
                          "flexDirection": "row",
                          "paddingHorizontal": 10,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <TextInput
                        editable={true}
                        multiline={true}
                        numberOfLines={4}
                        onChangeText={[Function]}
                        placeholderTextColor="#86939e"
                        style={
                          {
                            "color": "#242424",
                            "flex": 1,
                            "fontSize": 18,
                            "minHeight": 40,
                          }
                        }
                        testID="type-text"
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <Text
                      style={
                        {
                          "color": "#ff190c",
                          "fontSize": 12,
                          "margin": 5,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <View>
                <View>
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#242424",
                        "fontSize": 14,
                        "fontWeight": 600,
                        "marginBottom": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                    testID="field-label"
                  >
                    2. Organisation
                  </Text>
                  <View
                    style={
                      {
                        "paddingHorizontal": 10,
                        "width": "100%",
                      }
                    }
                    testID="RNE__Input__view-wrapper"
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomWidth": 0.5,
                          "borderColor": "grey",
                          "borderRadius": 5,
                          "borderWidth": 0.5,
                          "flexDirection": "row",
                          "paddingHorizontal": 10,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <TextInput
                        editable={true}
                        onChangeText={[Function]}
                        placeholderTextColor="#86939e"
                        style={
                          {
                            "color": "#242424",
                            "flex": 1,
                            "fontSize": 18,
                            "minHeight": 40,
                          }
                        }
                        testID="type-input"
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <Text
                      style={
                        {
                          "color": "#ff190c",
                          "fontSize": 12,
                          "margin": 5,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <View>
                <View>
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#242424",
                        "fontSize": 14,
                        "fontWeight": 600,
                        "marginBottom": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                    testID="field-label"
                  >
                    3. Household code
                  </Text>
                  <View
                    style={
                      {
                        "paddingHorizontal": 10,
                        "width": "100%",
                      }
                    }
                    testID="RNE__Input__view-wrapper"
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomWidth": 0.5,
                          "borderColor": "grey",
                          "borderRadius": 5,
                          "borderWidth": 0.5,
                          "flexDirection": "row",
                          "paddingHorizontal": 10,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <TextInput
                        editable={true}
                        multiline={true}
                        numberOfLines={4}
                        onChangeText={[Function]}
                        placeholderTextColor="#86939e"
                        style={
                          {
                            "color": "#242424",
                            "flex": 1,
                            "fontSize": 18,
                            "minHeight": 40,
                          }
                        }
                        testID="type-text"
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <Text
                      style={
                        {
                          "color": "#ff190c",
                          "fontSize": 12,
                          "margin": 5,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <View>
                <View
                  style={
                    {
                      "marginBottom": 25,
                    }
                  }
                >
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#242424",
                        "fontSize": 14,
                        "fontWeight": 600,
                        "marginBottom": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                    testID="field-label"
                  >
                    4. Are you willing to participate in the survey?
                  </Text>
                  <View
                    accessibilityRole="checkbox"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": false,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    checkedIcon="dot-circle-o"
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "borderColor": "#ffffff",
                        "margin": 5,
                        "marginBottom": 0,
                        "marginLeft": 10,
                        "marginRight": 10,
                        "padding": 10,
                        "paddingVertical": 5,
                      }
                    }
                    testID="RNE__CheckBox__Wrapper"
                    uncheckedIcon="circle-o"
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text />
                      <Text
                        accessibilityRole="text"
                        style={
                          {
                            "color": "#43484d",
                            "fontWeight": "normal",
                            "marginLeft": 10,
                            "marginRight": 10,
                          }
                        }
                        testID="RNE__CheckBox__Title"
                      >
                        Yes
                      </Text>
                    </View>
                  </View>
                  <View
                    accessibilityRole="checkbox"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": false,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    checkedIcon="dot-circle-o"
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "borderColor": "#ffffff",
                        "margin": 5,
                        "marginBottom": 0,
                        "marginLeft": 10,
                        "marginRight": 10,
                        "padding": 10,
                        "paddingVertical": 5,
                      }
                    }
                    testID="RNE__CheckBox__Wrapper"
                    uncheckedIcon="circle-o"
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text />
                      <Text
                        accessibilityRole="text"
                        style={
                          {
                            "color": "#43484d",
                            "fontWeight": "normal",
                            "marginLeft": 10,
                            "marginRight": 10,
                          }
                        }
                        testID="RNE__CheckBox__Title"
                      >
                        No
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <View>
                <View>
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#242424",
                        "fontSize": 14,
                        "fontWeight": 600,
                        "marginBottom": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                    testID="field-label"
                  >
                    5. Number between 1 to 9 (decimal allowed)
                  </Text>
                  <View
                    style={
                      {
                        "paddingHorizontal": 10,
                        "width": "100%",
                      }
                    }
                    testID="RNE__Input__view-wrapper"
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomWidth": 0.5,
                          "borderColor": "grey",
                          "borderRadius": 5,
                          "borderWidth": 0.5,
                          "flexDirection": "row",
                          "paddingHorizontal": 10,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <TextInput
                        editable={true}
                        keyboardType="numeric"
                        onChangeText={[Function]}
                        placeholderTextColor="#86939e"
                        style={
                          {
                            "color": "#242424",
                            "flex": 1,
                            "fontSize": 18,
                            "minHeight": 40,
                          }
                        }
                        testID="type-number"
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <Text
                      style={
                        {
                          "color": "#ff190c",
                          "fontSize": 12,
                          "margin": 5,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View>
    <View
      accessibilityRole="tablist"
      onLayout={[Function]}
      style={
        [
          false,
          {
            "flexDirection": "row",
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "Back",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-btn-back"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "marginHorizontal": 5,
                  "overflow": "hidden",
                  "paddingBottom": 10,
                  "paddingTop": 9,
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "grey",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 2,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "1/2",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-group-count"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "column",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "hsl(208, 8%, 63%)",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 8,
                  "textAlign": "center",
                }
              }
            >
              1/2
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "Next",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-btn-next"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "row-reverse",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "marginHorizontal": 5,
                  "overflow": "hidden",
                  "paddingBottom": 10,
                  "paddingTop": 9,
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "grey",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 2,
                  "textAlign": "center",
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
